app/components/error-wrapper.js

```diff
@@ -0,0 +1,50 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  classNames: ['error-wrapper', 'center-pseudo'],
+
+  // Map the HTTP status codes into an array or
+  // an empty array if there are no such status codes
+  httpStatusCodes: Ember.computed('model', function() {
+    let model = this.get('model');
+    if (model && model.hasOwnProperty('errors')) {
+      let errors = model.errors;
+      return errors.map(function(err) {
+        return err.status;
+      });
+    } else {
+      return [];
+    }
+  }),
+
+  is404: Ember.computed('httpStatusCodes', function() {
+    return this.get('httpStatusCodes').contains(404);
+  }),
+
+  errorClass: Ember.computed('is404', function() {
+    if(this.get('is404')) {
+      return 'warning';
+    } else {
+      return 'danger';
+    }
+  }),
+
+  background: Ember.inject.service(),
+
+  observeErrorClass: Ember.observer('errorClass', function() {
+    this.updateBackground();
+  }),
+
+  updateBackground: function() {
+    this.set('background.class', this.get('errorClass'));
+    this.get('background').updateBackgroundClass();
+  },
```

Comment:
Are you sure we need this. At a glance here, it looks like we could just use 

```JavaScript
classNames: ['error-wrapper', 'center-pseudo'],
classNameBindings:  'errorClass']
```

This will dynamically set the components class to the value of the `errorClass` property, in addition to assigning the static `error-wrapper` and `center-pseudo` classes.